examples/src/main/java/ai/djl/examples/inference/sr/SuperResolution.java

```diff
@@ -0,0 +1,131 @@
+
+import ai.djl.Application;
+import ai.djl.ModelException;
+import ai.djl.engine.Engine;
+import ai.djl.inference.Predictor;
+import ai.djl.modality.cv.Image;
+import ai.djl.modality.cv.ImageFactory;
+import ai.djl.modality.cv.util.NDImageUtils;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDArrays;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.training.util.ProgressBar;
+import ai.djl.translate.TranslateException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class SuperResolution {
+
+    private static final Logger logger = LoggerFactory.getLogger(SuperResolution.class);
+
+    private SuperResolution() {}
+
+    public static void main(String[] args) throws ModelException, TranslateException, IOException {
+        String imagePath = "src/test/resources/";
+        ImageFactory imageFactory = ImageFactory.getInstance();
+
+        List<Image> inputImages =
+                Arrays.asList(
+                        imageFactory.fromFile(Paths.get(imagePath + "fox.png")),
+                        imageFactory.fromFile(Paths.get(imagePath + "monkey.png")));
+        List<Image> enhancedImages = enhance(inputImages);
+
+        if (enhancedImages == null) {
+            logger.info("This example only works for TensorFlow Engine");
+        } else {
+            logger.info("Using TensorFlow Engine. {} images generated.", enhancedImages.size());
+            saveImages(inputImages, enhancedImages);
+        }
+    }
+
+    private static void saveImages(List<Image> input, List<Image> generated) throws IOException {
+        Path outputPath = Paths.get("build/output/super-res/");
+        Files.createDirectories(outputPath);
+
+        save(generated, "image", outputPath);
+        save(group(input, generated), "stitch", outputPath);
+
+        logger.info("Generated images have been saved in: {}", outputPath);
+    }
+
+    private static void save(List<Image> images, String name, Path path) throws IOException {
+        for (int i = 0; i < images.size(); i++) {
+            Path imagePath = path.resolve(name + i + ".png");
+            images.get(i).save(Files.newOutputStream(imagePath), "png");
+        }
+    }
+
+    private static List<Image> group(List<Image> input, List<Image> generated) {
+        NDList stitches = new NDList(input.size());
+
+        try (NDManager manager = Engine.getInstance().newBaseManager()) {
+            for (int i = 0; i < input.size(); i++) {
+                int width = input.get(i).getWidth();
+                int height = input.get(i).getHeight();
+
+                NDArray left = input.get(i).toNDArray(manager);
```

Comment:
I think we should move stitch part into translator

Reply:
Stitching the input image and output image is just for demo purposes so we can see the difference between the input and output. Do you still think this should go in the translator? It isn't technically part of postprocessing.